pkg/cosign/fetch.go

```diff
@@ -68,42 +70,39 @@ func FetchSignatures(ref name.Reference) ([]SignedPayload, *v1.Descriptor, error
 	}
 
 	signatures := make([]SignedPayload, len(descriptors))
-	var errs []error
-	var wg sync.WaitGroup
-	wg.Add(len(descriptors))
+
+	g, _ := errgroup.WithContext(context.Background())
+
 	for i, desc := range descriptors {
-		go func(i int, desc v1.Descriptor) {
-			defer wg.Done()
+		i, desc := i, desc
+		g.Go(func() error {
```

Comment:
We might want to bound this to some reasonable number of concurrent in flight requests?

Reply:
good idea! limited to the # of CPUs